include/warehouse_ros/impl/message_collection_impl.hpp

```diff
@@ -37,15 +37,19 @@
  * \author Bhaskara Marthi
  */
 
+#include <rcutils/allocator.h>
+#include <rmw/serialized_message.h>
+#include <rosidl_generator_cpp/traits.hpp>
+
 namespace warehouse_ros
 {
 template <class M>
 MessageCollection<M>::MessageCollection(MessageCollectionHelper::Ptr collection) : collection_(collection)
 {
-  typedef typename ros::message_traits::DataType<M> DataType;
-  const std::string datatype = DataType().value();
-  typedef typename ros::message_traits::MD5Sum<M> Md5;
-  const std::string md5 = Md5().value();
+  const std::string datatype = rosidl_generator_traits::data_type<M>();
+  // TODO: Get MD5Sum value
+  // typedef typename ros::message_traits::MD5Sum<M> Md5;
```

Comment:
I'm not sure if md5 will be supported and also I didn't see any other corresponding trait. But it seems like hashing is critical here. Couldn't we just hash the serialized message string using `std::hash` https://en.cppreference.com/w/cpp/utility/hash?

Reply:
Great idea! I found a method from `openssl` to get the md5sum string from the message type. But I think this is still not equal to the function on ros1 that returns the hash number of the message definition. I'm not sure if this is enough for the application.

Reply:
Looks good to me! Are you sure this is different? ROS1 also seems to hash the message type, but then again this check should actually compare if the message definition matches. For now this should work though, as this should really just protect against using mismatching message versions. Maybe we could throw a warning about this.